Mathlib/Combinatorics/Enumerative/DyckWord.lean

```diff
@@ -196,4 +221,142 @@ lemma two_mul_semilength_eq_length : 2 * p.semilength = p.toList.length := by
 
 end Semilength
 
+section FirstReturn
+
+variable (p) in
+/-- `p.firstReturn` is 0 if `p = 0` and the index of the `D` matching the initial `U` otherwise. -/
+def firstReturn : ℕ :=
+  (range p.toList.length).findIdx fun i ↦
+    (p.toList.take (i + 1)).count U = (p.toList.take (i + 1)).count D
+
+@[simp] lemma firstReturn_zero : firstReturn 0 = 0 := rfl
+
+include h in
+lemma firstReturn_pos : 0 < p.firstReturn := by
```

Comment:
Either write the corresponding `positivity` extension or
```suggestion
-- TODO: `positivity` extension
include h in
lemma firstReturn_pos : 0 < p.firstReturn := by
```

Reply:
All done.